gap/oper.gi

```diff
@@ -1365,6 +1365,37 @@ InstallMethod(DigraphDijkstra, "for a digraph, and a vertex",
 [IsDigraph, IsPosInt],
 {digraph, source} -> DIGRAPHS_DijkstraST(digraph, source, fail));
 
+InstallMethod(DigraphDijkstraSTWeights, "for a digraph, a vertex, a vertex, and a list of weights",
```

Comment:
I suggest using your new version `DigraphDijkstraSTWeights` as the code of `DIGRAPHS_DijkstraST`, with one change: the argument `weights` could be a function like:
~~~~
weights := function(u, v)
  return list_of_weights[u][v];
end;
~~~~
if a list of weights is given, and if no list of weights is given then we could do:
~~~~
weights := function(u, v)
  return 1;
end;
~~~~